### ISSUE TITLE
Duplicate `_main` symbol when linking abseil

### ISSUE THREAD

**Issue by mohsin** [google/nearby#1187]
I tried making an empty SwiftUI project and importing this but the build failed and gave me an error `1 duplicate symbol for architecture arm64`. On M1 mac so I'm thinking this might have to do with this? I thought it had to do with [this](https://stackoverflow.com/a/32740175/997147) but turning off common blocks did not help. Goal here is to build a sample macOS (Desktop) app and see if I can send a file from my Android phone to Mac using this library.

**Comment by bourdakos1** [comment]
Hi @mohsin, we just did a large refactor involving all the c++ namespaces, so this could be a regression. I’ll take a look and make sure it still builds on my M1 MacBook.

**Comment by bourdakos1** [comment]
Yea, its failing for me with the same error. I'll look into what broke and why our tests are still passing.

**Comment by mohsin** [comment]
Sure, thank you! Appreciate it.

**Comment by bourdakos1** [comment]
Okay, I think I've fixed the issue. You can test by changing the nearby branch from `master` to `cl/500537425`

**Comment by bourdakos1** [comment]
Fixed in https://github.com/google/nearby/pull/1189

**Comment by mohsin** [comment]
@bourdakos1 Yes, it is resolved. I did have to fork out and make one modification i.e. switch `exclude-main` to `main` as the [exclude-main branch](https://github.com/google/nearby/commit/bf88e3146f56aa15bf8fbf60096cf835a22a94d1) was missing on your fork. I also noticed that you added a Test for the same to the master branch of this repo but missed out merging `cl/500537425` which is the main fix that the tests are meant for. Thanks again for a swift fix. Appreciate it!
